### PR TITLE
olive-editor-nightly: Add manifest

### PR DESCRIPTION
**Pull request by igitur**
Another attempt to add Olive Editor Nightly. 

This uses nightly.link as a proxy service for Github Actions artifacts

Relates to #342

- [X] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tech189** [comment]
The hash in autoupdate can be removed, they don't provide the hash as an artifact in the Actions run: https://github.com/olive-editor/olive/actions/runs/2484266448#artifacts

Also you'll need to add to the pre_install script - the downloaded zip contains another zip inside.

Other maintainers might need to confirm that we can use the nightly link service. I don't think any other app uses it.

**Comment by igitur** [comment]
Sorry, this is still WIP. How do I make this a draft PR again?

**Comment by tech189** [comment]
No worries, the button should be on the right hand side: "Convert to draft"

**Comment by igitur** [comment]
Ah, it's a tiny little link, not a button. I totally missed it initially. Thanks

File: bucket/olive-editor-nightly.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "architecture": {
+        "64bit": {
+            "url": "https://nightly.link/olive-editor/olive/workflows/ci/master/Olive-5ef56c80-Windows-x86_64.zip.zip",
+            "hash": "f79d3816c1b9a13fa35c17f087d9bcea2b8f29febdc0e74acc5587e8e203b8a8"
+        }
+    },
+    "pre_install": [
+        "7z x \"$dir\\Olive*.zip\" -o\"$dir\\tmp\"",
+        "Move-Item -Path \"$dir\\tmp\\olive-editor\\*\" -Destination \"$dir\\\"",
+        "Remove-Item \"$dir\\tmp\" -Recurse",
```
Comment (tech189) [inline bucket/olive-editor-nightly.json:15]:
```suggestion
        "Expand-7zipArchive -Path \"$dir\\Olive*.zip\" -ExtractDir 'olive-editor' -Removal",
```
Can be simplified using the built-in Scoop function.

File: bucket/olive-editor-nightly.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        "'config.xml', 'layout', 'shortcuts' | ForEach-Object { New-Item \"$dir\\$_\" -ItemType File | Out-Null }",
```
Comment (tech189) [inline bucket/olive-editor-nightly.json:16]:
```
---------------------------
Error loading settings
---------------------------
Failed to load application settings. This session will use defaults.

Premature end of document.
---------------------------
OK   
---------------------------
```
Getting this error, maybe you can fill in config.xml with an example setting?

Comment (igitur) [reply bucket/olive-editor-nightly.json]: Ok done.

**Comment by tech189** [comment]
It installs fine but when uninstalling, it can't find `shortcuts`:
```
Unlinking ~\scoop\apps\olive-editor-nightly\current
Get-Item: C:\Users\me\scoop\apps\scoop\current\lib\install.ps1:1197:23
Line |
1197 |              $source = Get-Item "$dir\$source"
     |                        ~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path 'C:\Users\me\scoop\apps\olive-editor-nightly\5ef56c80\shortcuts' because it
     | does not exist.
```
Does this happen for you when uninstalling? What is the shortcuts file for?

**Comment by igitur** [comment]
> Does this happen for you when uninstalling? What is the shortcuts file for?

To be honest, I'm not sure. I based this PR on what used to be in https://github.com/ScoopInstaller/Nightlies/pull/5 (doesn't exist anymore). I'll see if I can find any references to `shortcuts` in the Olive source code.

**Comment by igitur** [comment]
Olive Editor Nightly breaks the link to the symlinked files in the `persist` directory whenever I save preferences or shortcuts. Looking at the source code, I see that, when saving, it creates a new file and moves that over the existing file. There's not much we can do about that. This is a nightly version, so we can expects quirks anyway.

Thanks for the changes. I'm happy with them as they are.

**Comment by tech189** [comment]
In my changes there is a fix to that broken persisted config problem: https://github.com/ScoopInstaller/Versions/pull/556/files#diff-e5b1ad718863fadee9710f166c95883ab5942744ff07fd4c574ea8003693c61aR20-R26. It copies the config manually to the persist folder on uninstallation/update to new nightly build.

**Review by tech189 — APPROVED** [review]

**Comment by tech189** [comment]
@issaclin32 this is ready to go, just wanted your opinion on using nightly.link to get the download url?

File: bucket/olive-editor-nightly.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            "Olive Editor Nightly"
+        ]
+    ],
+    "persist": [
+        "config.xml",
+        "shortcuts"
+    ],
+    "checkver": {
+        "url": "https://nightly.link/olive-editor/olive/workflows/ci/master",
+        "regex": ">Olive-(?<version>[\\d\\w]{8})-Windows-x86_64\\.zip<\\/a>"
```
Comment (issaclin32) [inline bucket/olive-editor-nightly.json:40]:
```suggestion
        "regex": ">Olive-([\\d\\w]{8})-Windows-x86_64\\.zip</a>"
```
The regex can be simplfied. The 1st match group will be **version**, and `/` does not need escape in .NET variant of regex.

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**